artwork/Modules/Project/Services/ProjectService.php

```diff
@@ -554,4 +557,71 @@ public function pinnedProjects(): Collection
     {
         return $this->projectRepository->pinnedProjects();
     }
+
+    public function attachManagementUsersWithoutSelf(Project $project, array $userIds): void
+    {
+        $usersToAttach = collect($userIds)->filter(fn($user) => $user !== Auth::id())
+            ->mapWithKeys(fn($user) => [$user => [
+                'access_budget' => false,
+                'is_manager' => true,
+                'can_write' => false,
+                'delete_permission' => false
+            ]]);
+
+        $project->users()->attach($usersToAttach);
+    }
+
+    public function attachManagementUsersWithSelf(Project $project, array $userIds): void
+    {
+        $usersToAttach = collect($userIds)
+            ->mapWithKeys(fn($user) => [$user => [
+                'access_budget' => false,
+                'is_manager' => true,
+                'can_write' => false,
+                'delete_permission' => false
+            ]]);
+
+        $project->users()->attach($usersToAttach);
+    }
+
+    public function attachSelfToProject(Project $project, bool $isManager): void
+    {
+        $project->users()->attach(Auth::id(), [
+            'access_budget' => false,
+            'is_manager' => $isManager,
+            'can_write' => false,
+            'delete_permission' => false
+        ]);
+    }
+
+    public function syncCategories(Project $project, array $categories): void
+    {
+        $project->categories()->sync($categories);
+    }
+
+    public function syncGenres(Project $project, array $genres): void
+    {
+        $project->genres()->sync($genres);
+    }
+
+    public function syncSectors(Project $project, array $sectors): void
+    {
+        $project->sectors()->sync($sectors);
+    }
+
+    public function detachingManagementUsers(Project $project, bool $detachingAll = false, array $userIds = []): void
```

Comment:
SHOULD: detachManagementUsers

detaching suggeriert dass es sich um eine Funktion handelt die einen bool liefert.